src/buffer/input_view.rs

```diff
@@ -326,7 +336,7 @@ impl State {
     }
 
     pub fn focus(&self) -> Task<Message> {
-        text_input::focus(self.input_id.clone())
+        text_input::is_focused(self.input_id.clone()).map(Message::Focus)
```

Comment:
You don't need `Message::Focus`, you can `Task::then` to chain `text_input::focus` after this using the conditional from the previous operation